chunky/src/java/se/llbit/chunky/renderer/scene/SkyCache.java

```diff
@@ -0,0 +1,162 @@
+/* Copyright (c) 2021 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer.scene;
+
+import org.apache.commons.math3.util.FastMath;
+import se.llbit.math.ColorUtil;
+import se.llbit.math.QuickMath;
+import se.llbit.math.Ray;
+import se.llbit.math.Vector3;
+
+import static java.lang.Math.PI;
+
+public class SkyCache {
+  // Sky texture array, HSV colors
+  private double[][][] skyTexture;
+
+  // Default resolution is 1024x1024. Should be more than enough for any simulated sky.
+  private int skyResolution = 1024;
```

Comment:
8 
![image](https://user-images.githubusercontent.com/3740917/107713483-40695700-6cc3-11eb-929e-f782e5ebdb04.png)

16
![image](https://user-images.githubusercontent.com/3740917/107713506-4bbc8280-6cc3-11eb-962f-b90a84dc0b53.png)

32
![image](https://user-images.githubusercontent.com/3740917/107713538-5c6cf880-6cc3-11eb-9350-bc5064aa5186.png)

64
![image](https://user-images.githubusercontent.com/3740917/107713588-7eff1180-6cc3-11eb-9788-5af32bf76f7c.png)

128
![image](https://user-images.githubusercontent.com/3740917/107713623-8faf8780-6cc3-11eb-8d5e-03681e5c065c.png)

While a cache size of 32 is too low and 64 appears to be just right, I think going for 128 as the safer option would be best. Performance with the current default of 1024 should give us the same performance as previous versions Preetham.

Reply:
A cache size of 128 seems good. I can change the default to that. It should improve performance a lot, especially when it has to rebuild the cache on Nishita sky.

Reply:
Changed cache size to 128 in b3fb88d